src/sentry/utils/raven.py

```diff
@@ -60,69 +63,29 @@ def send(self, **kwargs):
         if not is_current_event_safe():
             return
 
-        # Force raven-python encoding of this event, so that edge cases like
-        # non JSON-serializable objects are handled.
-        kwargs = self.decode(self.encode(kwargs))
-
-        from sentry import tsdb
-        from sentry.coreapi import ClientApiHelper
-        from sentry.event_manager import EventManager
-        from sentry.models import Project
-
-        helper = ClientApiHelper(
-            agent='raven-python/%s (sentry %s)' % (raven.VERSION, sentry.VERSION),
-            project_id=settings.SENTRY_PROJECT,
-            version=self.protocol_version,
+        from sentry.web.api import StoreView
+        internal_config = RemoteConfig.from_string(settings.SENTRY_INTERNAL_DSN)
```

Comment:
@mattrobenolt Something like this? 

In this case how would we populate this value? just hardcode it in the settings here? or have it be some sort of environment variable? 

The other thing I guess is that `SENTRY_PROJECT` becomes a little redundant if the project ID is encoded in the DSN, but I don't want to go and change all of that.

Reply:
Sorry, to be more explicit, I meant, we point to a key ID.

So the code would be something like this:

```python
if settings.SENTRY_PROJECT_KEY is not None:
    key = ProjectKey.objects.get(id=settings.SENTRY_PROJECT_KEY)
else:
    key = ProjectKey.objects.filter(project_id=settings.SENTRY_PROJECT)[0]
```

Then build the DSN from the key and project id. This way it's not required to be set, and we default it to just the first key under this project. (Obviously with some error handling if there aren't any keys).

This way we can now point the key to an explicit one for us internally and things in theory should work just fine on new installs.

These bits can then just be passed into the `get_auth_header` call below with just `key.{public,private}_key`. So we can bypass this `RemoteConfig.from_string` call entirely.

Reply:
Might even be worth wrapping this in some cache.

Reply:
There's also `ProjectKey.get_default(project)` that we can probably just use if they don't have an id specified.